plugins/source/aws/resources/services/apigatewayv2/apis_fetch.go

```diff
@@ -2,88 +2,144 @@ package apigatewayv2
+		Service:   string(client.ApigatewayService),
+		Region:    cl.Region,
+		AccountID: "",
+		Resource:  fmt.Sprintf("/apis/%s/integrations/%s", aws.ToString(p.ApiId), aws.ToString(r.IntegrationId)),
+	}.String())
 }
-func resolveApiIntegrationResponseArn() schema.ColumnResolver {
-	return client.ResolveARNWithRegion(client.ApigatewayService, func(resource *schema.Resource) ([]string, error) {
-		r := resource.Item.(types.IntegrationResponse)
-		i := resource.Parent.Item.(types.Integration)
-		api := resource.Parent.Parent.Item.(types.Api)
-		return []string{apiIDPart, *api.ApiId, apiIntegrationIDPart, *i.IntegrationId, "integrationresponses", *r.IntegrationResponseId}, nil
-	})
+
+func resolveApiIntegrationResponseArn(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	cl := meta.(*client.Client)
+	r := resource.Item.(types.IntegrationResponse)
+	i := resource.Parent.Item.(types.Integration)
+	api := resource.Parent.Parent.Item.(types.Api)
+	return resource.Set(c.Name, arn.ARN{
+		Partition: cl.Partition,
+		Service:   string(client.ApigatewayService),
+		Region:    cl.Region,
+		AccountID: "",
+		Resource:  fmt.Sprintf("/apis/%s/integrations/%s/integrationresponses/%s", aws.ToString(api.ApiId), aws.ToString(i.IntegrationId), aws.ToString(r.IntegrationResponseId)),
+	}.String())
 }
-func resolveApiModelArn() schema.ColumnResolver {
-	return client.ResolveARNWithRegion(client.ApigatewayService, func(resource *schema.Resource) ([]string, error) {
-		r := resource.Item.(types.Model)
-		p := resource.Parent.Item.(types.Api)
-		return []string{apiIDPart, *p.ApiId, "models", *r.ModelId}, nil
-	})
+
+func resolveApiModelArn(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	cl := meta.(*client.Client)
+	r := resource.Item.(types.Model)
+	p := resource.Parent.Item.(types.Api)
+	return resource.Set(c.Name, arn.ARN{
+		Partition: cl.Partition,
+		Service:   string(client.ApigatewayService),
+		Region:    cl.Region,
+		AccountID: "",
+		Resource:  fmt.Sprintf("/apis/%s/models/%s", aws.ToString(p.ApiId), aws.ToString(r.ModelId)),
+	}.String())
 }
-func resolveApiRouteArn() schema.ColumnResolver {
-	return client.ResolveARNWithRegion(client.ApigatewayService, func(resource *schema.Resource) ([]string, error) {
-		r := resource.Item.(types.Route)
-		p := resource.Parent.Item.(types.Api)
-		return []string{apiIDPart, *p.ApiId, apiRouteIDPart, *r.RouteId}, nil
-	})
+
+func resolveApiRouteArn(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	cl := meta.(*client.Client)
+	r := resource.Item.(types.Route)
+	p := resource.Parent.Item.(types.Api)
+	return resource.Set(c.Name, arn.ARN{
+		Partition: cl.Partition,
+		Service:   string(client.ApigatewayService),
+		Region:    cl.Region,
+		AccountID: "",
+		Resource:  fmt.Sprintf("/apis/%s/resolveApiRouteResponseArn/%s", aws.ToString(p.ApiId), aws.ToString(r.RouteId)),
```

Comment:
The `resolveApiRouteResponseArn` part here doesn't look right 🤔

Reply:
Nice catch!!